components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -99,6 +101,12 @@ class GeckoEngineSession(
      */
     private fun createNavigationDelegate() = object : GeckoSession.NavigationDelegate {
         override fun onLocationChange(session: GeckoSession?, url: String) {
+            // Ignore initial load of about:blank (see https://github.com/mozilla-mobile/android-components/issues/403)
+            if (initialLoad && url == ABOUT_BLANK) {
+                return
+            }
+            initialLoad = false
```

Comment:
We set `initialLoad` to `false` in onLocationChange and onSecurityChange. Is it guaranteed that only one of those happens? Will it never happen that one of them gets called, we set `initialLoad` to `false` and then do not ignore `about:blank` in the other callback?